lib/metrics/amplitude.js

```diff
@@ -165,23 +173,56 @@ module.exports = log => {
 
     return P.resolve()
   }
-}
 
-function getFromToken (request, key) {
-  if (request.auth.credentials) {
-    return request.auth.credentials[key]
+  function getFromToken (request, key) {
+    if (request.auth.credentials) {
+      return request.auth.credentials[key]
+    }
   }
-}
 
-function getFromMetricsContext (metricsContext, key, request, payloadKey) {
-  return metricsContext[key] ||
-    (request.payload.metricsContext && request.payload.metricsContext[payloadKey])
-}
+  function getFromMetricsContext (metricsContext, key, request, payloadKey) {
+    return metricsContext[key] ||
+      (request.payload.metricsContext && request.payload.metricsContext[payloadKey])
+  }
+
+  function mapEventProperties (group, request, data, metricsContext) {
+    return Object.assign({
+      service: data.service || request.payload.service || request.query.service
+    }, EVENT_PROPERTIES[group](request, data, metricsContext))
+  }
+
+  function mapUserProperties (group, request, data, metricsContext) {
+    const { browser, browserVersion, os } = request.app.ua
+    return Object.assign({
+      flow_id: getFromMetricsContext(metricsContext, 'flow_id', request, 'flowId'),
+      sync_device_count: data.devices && data.devices.length,
+      ua_browser: browser,
+      ua_version: browserVersion,
+      ua_os: os,
+      user_country: getLocationProperty(data, 'country'),
+      user_locale: getLocale(request),
+      user_state: getLocationProperty(data, 'state'),
+    }, getService(request))
+  }
+
+  function getLocale (request) {
+    return request.app.locale || undefined
+  }
 
-function mapEventProperties (group, request, data, metricsContext) {
-  return Object.assign({
-    service: data.service || request.query.service || request.payload.service
-  }, EVENT_PROPERTIES[group](request, data, metricsContext))
+  function getLocationProperty (data, key) {
+    return (data.location && data.location[key]) || undefined
+  }
+
+  function getService (request) {
+    const service = SERVICES.get(request.payload.service || request.query.service)
+    if (service) {
+      return {
+        '$append': {
+          fxa_services_used: service
+        }
+      }
+    }
+  }
 }
 
 function mapEmailEventProperties (request, data) {
```

Comment:
`mapEmailEventProperties` has to stay outside the closure all on its own, because it's referred to by `EVENT_PROPERTIES`.